src/panels/SkyBrowserPanel/hooks.ts

```diff
@@ -0,0 +1,223 @@
+    lowPrecisionEqualArray
+  );
+  return color;
+}
+
+export function useSelectedBrowserColorString(): string | undefined {
+  const color = useSelectedBrowserColor();
+  return color ? (`rgb(${color.join(',')})` as string) : undefined;
+}
+
+export function useSelectedBrowserRadius(): number | undefined {
+  const radius = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.borderRadius,
+    lowPrecisionEqual()
+  );
+  return radius;
+}
+
+export function lowPrecisionEqualMatrix(
+  lhs: number[][] | undefined,
+  rhs: number[][] | undefined
+) {
+  if (lhs === undefined && rhs === undefined) {
+    return true;
+  }
+  if (lhs === undefined || rhs === undefined) {
+    return false;
+  }
+  if (lhs.length !== rhs.length) {
+    return false;
+  }
+  return lhs.every((value, i) => lowPrecisionEqualArray(value, rhs[i]));
+}
+
+export function equalArray<T>(lhs: T[] | undefined, rhs: T[] | undefined) {
+  if (lhs === undefined && rhs === undefined) {
+    return true;
+  }
+  if (lhs === undefined || rhs === undefined) {
+    return false;
+  }
+  if (lhs.length !== rhs.length) {
+    return false;
+  }
+  return lhs.every((value, i) => value === rhs[i]);
+}
+
+export function lowPrecisionEqualArray(
+  lhs: number[] | undefined,
+  rhs: number[] | undefined
+) {
+  if (lhs === undefined && rhs === undefined) {
+    return true;
+  }
+  if (lhs === undefined || rhs === undefined) {
+    return false;
+  }
+  if (lhs.length !== rhs.length) {
+    return false;
+  }
+
+  return lhs?.every((value, i) => lowPrecisionEqual()(value, rhs[i]));
+}
+
+export function lowPrecisionEqual(epsilon: number = 1e-3) {
+  return (lhs: number | undefined, rhs: number | undefined) => {
+    if (lhs === undefined && rhs === undefined) {
+      return true;
+    }
+    if (lhs === undefined || rhs === undefined) {
+      return false;
+    }
+    return Math.abs(lhs - rhs) < epsilon;
+  };
+}
+
+export function useSelectedBrowserCoords() {
+  const ra = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.ra,
+    lowPrecisionEqual(1e-6)
+  );
+  const dec = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.dec,
+    lowPrecisionEqual(1e-6)
+  );
+
+  const roll = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.roll,
+    lowPrecisionEqual(1e-6)
+  );
+  return { ra, dec, roll };
+}
+
+export function useSelectedBrowserFov() {
+  const fov = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.fov,
+    lowPrecisionEqual(1e-6)
+  );
+
+  return fov;
+}
+
+export function useSkyBrowserCartesianDirection() {
+  const cartesianDirection = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.cartesianDirection,
+    lowPrecisionEqualArray
+  );
+  return cartesianDirection;
+}
+
+export function useSkyBrowserColors() {
+  const browserColors = useAppSelector(
+    (state) => state.skybrowser.browserColors,
+    lowPrecisionEqualMatrix
+  );
+  return browserColors;
```

Comment:
```suggestion
  return useAppSelector(
    (state) => state.skybrowser.browserColors,
    lowPrecisionEqualMatrix
  );
```

and for the final hooks in this file.. you get it by now :)